### PR TITLE
compare recovered signer with tx.from

### PR DESCRIPTION
**Pull request by i-m-aditya**

### CODE REVIEW

File: src/executor/mod.rs [hunk 1]
```diff
@@ -46,12 +54,105 @@ impl EvmExecutor {
...
+            let result = secp.recover_ecdsa(&msg, &signature);
+
+            let secp = Secp256k1::new();
+            let result = secp.recover_ecdsa(&msg, &signature);
+
+            match result {
+                Ok(public_key) => {
+                    // Convert the public key to an Ethereum address
+                    let public_key_bytes = public_key.serialize_uncompressed();
+                    let hash = keccak256(&public_key_bytes[1..]); // Skip the first byte (0x04) which indicates uncompressed key
```
[the remark this answers is not in the document]
Comment (lispc) [reply src/executor/mod.rs:108]: will you use tiny-keccak?

**Comment by lispc** [comment]
the PR brings in too many lines of codes...   Is there some libs we can use?

I read the eth_tx.recover_from implemention, and thinks it is not too "patch friendly".   Maybe you can visit similar features (recover addr from sig) from alloy-rs?  if needed, we could switch to alloy-rs for (some?) crates?

File: src/executor/mod.rs [hunk 2]
```diff
@@ -52,6 +52,17 @@ impl EvmExecutor {
                 idx,
                 l2_trace.base_fee_per_gas(),
             );
+
+            let recovered_address = eth_tx.recover_from().unwrap();
```
[the remark this answers is not in the document]
Comment (roynalnaruto) [reply src/executor/mod.rs:56]: For reference, this is being called:
https://docs.rs/ethers-core/latest/ethers_core/types/struct.Signature.html#method.recover

**Review by roynalnaruto — CHANGES_REQUESTED** [review]
As I mentioned, we will need some refactoring before we merge this PR, especially related to error handling in SBV. Before that, you cannot really test the malicious case (incorrect signature), as your code would panic and this is something that we don't want.

File: testdata/mainnet_blocks/5224657u.json
Comment (roynalnaruto) [inline testdata/mainnet_blocks/5224657u.json:1]: Was this committed by mistake? Not sure what the `5224657u` is?

File: src/executor/block_errors.rs [hunk 3]
```diff
@@ -0,0 +1,9 @@
+/// Transaction validation errors
+#[derive(thiserror_no_std::Error, Debug, Clone, PartialEq, Eq)]
+pub enum BlockVerificationError {
+    #[error("failed to recover sender for transaction")]
+    SenderRecoveryError,
```
Comment (roynalnaruto) [inline src/executor/block_errors.rs:5]: Based on the following review comment, we will need to propagate the `SignatureError` (from `ethers`) as well. So we may need something like:

```rust
#[error(transparent)]
SenderRecovery(#[from] ethers_core::types::SignatureError),
#[error("signature verification error: from={0}, recovered={}")]
SenderMismatch(Address, Address)
```

**Review by roynalnaruto — CHANGES_REQUESTED** [review]
A few more recommendations

File: testdata/mainnet_blocks/5224657u.json
Comment (i-m-aditya) [reply testdata/mainnet_blocks/5224657u.json:1]: Previous 5224657.json doesn't contain the codes, so for testing I used 5224657u.json.

File: src/executor/block_errors.rs (hunk 3 above)
Comment (roynalnaruto) [reply src/executor/block_errors.rs:5]: But as I mentioned, its best to do this refactoring in a separate PR. I already have a WIP PR for this. So we can keep #19 on hold until we get the refactoring merged.

File: testdata/mainnet_blocks/5224657u.json
Comment (lispc) [reply testdata/mainnet_blocks/5224657u.json:1]: @lightsing will update the json files to latest format soon. Let merge his PR first.

File: src/executor/block_errors.rs (hunk 3 above)
Comment (i-m-aditya) [reply src/executor/block_errors.rs:5]: @roynalnaruto can you share your WIP PR?

Comment (roynalnaruto) [reply src/executor/block_errors.rs:5]: #24

**Comment by roynalnaruto** [comment]
Depends on #24

**Comment by lispc** [comment]
Again you added a new json, is that needed?

File: testdata/mainnet_blocks/5224657u.json
Comment (lightsing) [reply testdata/mainnet_blocks/5224657u.json:1]: check

**Comment by i-m-aditya** [comment]
> Again you added a new json, is that needed?

Removed.

**Review by lispc — APPROVED** [review]

**Pull request merged**